velox/connectors/clp/ClpConnector.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "clp_s/TimestampPattern.hpp"
+
+#include "velox/connectors/clp/ClpConnector.h"
+#include "velox/connectors/clp/ClpDataSource.h"
+
+namespace facebook::velox::connector::clp {
+ClpConnector::ClpConnector(
+    const std::string& id,
+    std::shared_ptr<const config::ConfigBase> config)
+    : Connector(id), config_(std::make_shared<ClpConfig>(config)) {}
```

Comment:
```suggestion
    : Connector(id), config_(std::make_shared<ClpConfig>(std::move(config))) {}
```
I think here the ownership should be transferred

Reply:
Not sure if we should keep it consistent with the hive connector, and it's pretty cheap to copy a shared pointer.

Reply:
let @kirkrodrigues decide

Reply:
Seems harmless to move, right? Even if copying is cheap, since the pointer is already passed by value, ideally it should be moved, right?

Reply:
But we cannot guarantee the caller should not use `config` afterward. Also Hive connector also doesn't use move

Reply:
Moving parameters passed by value is a common pattern in C++, right? So if a user is modifying the code, they should be able to figure out they need to change this line. Also, the Hive connector is using `config` in two places so that might be one of the reasons they avoided `move`.